SystemInformer/mainwnd.c

```diff
@@ -284,46 +284,161 @@ PPH_STRING PhMwpInitializeWindowTitle(
-        PhAppendStringBuilder2(&stringBuilder, L"-");
-        break;
-    case KphLevelMin:
-        PhAppendStringBuilder2(&stringBuilder, L"--");
-        break;
+        PhInitializeStringBuilder(&stringBuilder, 100);
+        PhAppendStringBuilder2(&stringBuilder, PhApplicationName);
+
+        if (myUserAndDomain = PhGetSidFullName(PhGetOwnTokenAttributes().TokenSid, TRUE, NULL))
+        {
+            PhAppendStringBuilder2(&stringBuilder, L" [");
+            PhAppendStringBuilder(&stringBuilder, &myUserAndDomain->sr);
+            PhAppendCharStringBuilder(&stringBuilder, L']');
+            PhDereferenceObject(myUserAndDomain);
+        }
+
+        switch (KphLevelEx(FALSE))
+        {
+        case KphLevelMax:
+            PhAppendStringBuilder2(&stringBuilder, L"++");
+            break;
+        case KphLevelHigh:
+            PhAppendStringBuilder2(&stringBuilder, L"+");
+            break;
+        case KphLevelMed:
+            PhAppendStringBuilder2(&stringBuilder, L"~");
+            break;
+        case KphLevelLow:
+            PhAppendStringBuilder2(&stringBuilder, L"-");
+            break;
+        case KphLevelMin:
+            PhAppendStringBuilder2(&stringBuilder, L"--");
+            break;
+        }
+
+        if (PhGetOwnTokenAttributes().ElevationType == TokenElevationTypeFull)
+            PhAppendStringBuilder2(&stringBuilder, L" (Administrator)");
     }
+    else
+    {
+        //
+        // Verbose Developer Window Title ON
+        //
+
+        // 1. Starting Text
+        //
+        PhInitializeStringBuilder(&stringBuilder, 200);
+        PhAppendStringBuilder2(&stringBuilder, L"SI");
+
+        // 2. PID
+        //
+        PhAppendFormatStringBuilder(&stringBuilder, L" - %lu ", HandleToUlong(NtCurrentProcessId()));
 
-    if (PhGetOwnTokenAttributes().ElevationType == TokenElevationTypeFull)
-        PhAppendStringBuilder2(&stringBuilder, L" (Administrator)");
+        // 3. Username
+        //
+        BOOL isRunningAsSYSTEMUser = FALSE;
+        PSID mySID = PhGetOwnTokenAttributes().TokenSid;
+        if (myUserAndDomain = PhGetSidFullName(mySID, TRUE, NULL))
+        {
+            isRunningAsSYSTEMUser = RtlEqualSid(mySID, (PSID)&PhSeLocalSystemSid);
+
+            PhAppendStringBuilder2(&stringBuilder, L"- ");
+            PhAppendStringBuilder(&stringBuilder, &myUserAndDomain->sr);
+            PhAppendCharStringBuilder(&stringBuilder, L' ');
+            PhDereferenceObject(myUserAndDomain);
+        }
+
+        // 4. Elevated or not  (IL level string)
+        // 
+        BOOL isHighILOrGreater = FALSE;
+        MANDATORY_LEVEL integrityLevel;
+        PWSTR integrityString;
+        if (NT_SUCCESS(PhGetTokenIntegrityLevel(PhGetOwnTokenAttributes().TokenHandle, &integrityLevel, &integrityString)))
+        {
+            isHighILOrGreater = (integrityLevel >= MandatoryLevelHigh);
+        }
+        PhAppendFormatStringBuilder(&stringBuilder, L"   |   %ls IL", integrityString);
```

Comment:
There's lots of white space here. Is it really bringing value to have `L"   |   "` separators instead of just `L" | "`?